core/src/main/scala/org/elasticmq/actor/queue/MessageQueue.scala

```diff
@@ -126,7 +135,9 @@ object MessageQueue {
       messageQueue += message
     }
 
-    override def byId: Map[String, InternalMessage] = messagesById.toMap
```

Comment:
`messagesById.toMap` triggered unnecessary copy which increased the complexity to O(n)